File: system/ulib/ddk/protocol/input.c
// Copyright 2016 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <ddk/protocol/input.h>

const boot_kbd_report_t report_err_rollover = { .modifier = 1, .usage = {1, 1, 1, 1, 1, 1 } };

File: src/sys/fuzzing/fuzzctl/testing/src/input.rs
// Copyright 2022 The Fuchsia Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

use {
    anyhow::{Context as _, Result},
    std::fs,
    std::path::Path,
};

/// Verifies that the input was actually written and matches its expected contents.
pub fn verify_saved<P: AsRef<Path>>(saved: P, data: &[u8]) -> Result<()> {
    let saved = saved.as_ref();
    let actual =
        fs::read(saved).with_context(|| format!("failed to read '{}'", saved.to_string_lossy()))?;
    assert_eq!(actual, data);
    Ok(())
}
